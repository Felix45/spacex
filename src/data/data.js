const ships = [
  {
    id: 1,
    rocket_name: 'Falcon1',
    description: ' Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!',
    flickr_images: 'https://i.imgur.com/DaCfMsj.jpg',
  },
  {
    id: 2,
    rocket_name: 'Falcon9',
    description: ' Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!',
    flickr_images: 'https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg',
  },
  {
    id: 3,
    rocket_name: 'Starship',
    description: ' Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!',
    flickr_images: 'https://live.staticflickr.com/65535/48954138962_ee541e6755_b.jpg',
  },
];

export default ships;
