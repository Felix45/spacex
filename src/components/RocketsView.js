import React, { useState } from 'react';
import Container from 'react-bootstrap/Container';
import RocketList from './RocketList';
import { ships } from '../data/mock';

const RocketsView = () => {
  const [rockets] = useState(ships);

  return (
    <Container fluid>
      <RocketList rockets={rockets} />
    </Container>
  );
};

export default RocketsView;
