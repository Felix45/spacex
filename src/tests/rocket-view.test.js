import React from 'react';
import renderWithProviders from './test-utils';
import RocketsView from '../components/RocketsView';
import { rockets } from '../data/mock';

describe('RocketsView component', () => {
  const { asFragment, baseElement } = renderWithProviders(<RocketsView />, {
    preloadedState: {
      rockets: {
        rockets,
      },
    },
  });

  const firstRender = asFragment();

  it('Contains all the rockets from the mock data', () => {
    const renderedRockets = baseElement.getElementsByClassName('row');

    expect(renderedRockets.length).toBe(rockets.length);
  });

  it('Renders correctly', () => {
    expect(firstRender).toMatchSnapshot();
  });
});
