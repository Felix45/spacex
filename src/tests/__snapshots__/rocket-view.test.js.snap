// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketsView component Renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="my-5 row"
    >
      <div
        class="col-lg-3 col-md-6"
      >
        <div
          class="card"
        >
          <img
            alt="Falcon1"
            class="card-img-top"
            src="h"
          />
        </div>
      </div>
      <div
        class="col-lg-9 col-md-6"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              Falcon1
            </div>
            <p
              class="card-text"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!
            </p>
            <button
              class="btn btn-primary"
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="my-5 row"
    >
      <div
        class="col-lg-3 col-md-6"
      >
        <div
          class="card"
        >
          <img
            alt="Falcon9"
            class="card-img-top"
            src="h"
          />
        </div>
      </div>
      <div
        class="col-lg-9 col-md-6"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              Falcon9
            </div>
            <p
              class="card-text"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!
            </p>
            <button
              class="btn btn-primary"
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="my-5 row"
    >
      <div
        class="col-lg-3 col-md-6"
      >
        <div
          class="card"
        >
          <img
            alt="Starship"
            class="card-img-top"
            src="h"
          />
        </div>
      </div>
      <div
        class="col-lg-9 col-md-6"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              Starship
            </div>
            <p
              class="card-text"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!
            </p>
            <button
              class="btn btn-primary"
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
